common/math/math.go

```diff
@@ -34,3 +46,201 @@ func RoundFloat(x float64, prec int) float64 {
 	pow := math.Pow(10, float64(prec))
 	return math.Round(x*pow) / pow
 }
+
+// CompoundAnnualGrowthRate Calculates CAGR.
+// Using years, intervals per year would be 1 and number of intervals would be the number of years
+// Using days, intervals per year would be 365 and number of intervals would be the number of days
+func CompoundAnnualGrowthRate(openValue, closeValue, intervalsPerYear, numberOfIntervals float64) (float64, error) {
+	if numberOfIntervals == 0 {
+		return 0, errCAGRNoIntervals
+	}
+	if openValue == 0 {
+		return 0, errCAGRZeroOpenValue
+	}
+	k := math.Pow(closeValue/openValue, intervalsPerYear/numberOfIntervals) - 1
+	return k * 100, nil
+}
+
+// CalmarRatio is a function of the average compounded annual rate of return versus its maximum drawdown.
+// The higher the Calmar ratio, the better it performed on a risk-adjusted basis during the given time frame, which is mostly commonly set at 36 months
+func CalmarRatio(highestPrice, lowestPrice, average, riskFreeRateForPeriod float64) (float64, error) {
+	if highestPrice == 0 {
+		return 0, errCalmarHighest
+	}
+	drawdownDiff := (highestPrice - lowestPrice) / highestPrice
+	if drawdownDiff == 0 {
+		return 0, nil
+	}
+	return average - riskFreeRateForPeriod/drawdownDiff, nil
+}
+
+// InformationRatio The information ratio (IR) is a measurement of portfolio returns beyond the returns of a benchmark,
+// usually an index, compared to the volatility of those returns.
+// The benchmark used is typically an index that represents the market or a particular sector or industry.
+func InformationRatio(values, benchmarkRates []float64, averageValues, averageComparison float64) (float64, error) {
+	if len(benchmarkRates) == 1 {
+		for i := range values {
+			if i == 0 {
+				continue
+			}
+			benchmarkRates = append(benchmarkRates, benchmarkRates[0])
```

Comment:
If we only have a benchmark rate of say 0.03 per interval, this just means it will compare it against all points of data without needing to create a slice of 0.03 at len(intervals) before calling the function. It looks like the figure here is incorrect, but I don't think I'll be removing that code

Reply:
benchmark rate of 0.03 represents 3% per year?. If your candle is 1 day candle u need benchmark rate/365 rather than every candle having a benchmark rate of 0.03. Unless you intend to input a value already adjusted for the interval

Reply:
It was an example figure 😄 Meaning a standard benchmark per interval as I said

Reply:
I agree with your original assessment now anyway :D and will just require the user to provide the full slice themselves